src/com/clouway/validator/validator.js

```
@@ -0,0 +1,39 @@
+/**
+ * @author Stanislava Kaukova(sisiivanovva@gmail.com)
+ */
+
+function validateMatching(fieldId, regex) {
+    var value = document.getElementById(fieldId).value;
+
+    if (!regex.test(value)) {
+        displayError(fieldId);
+    } else {
+        hiddenError(fieldId);
+    }
+}
+
+function inRange(fieldId, minValue, maxValue) {
+    var value = document.getElementById(fieldId).value;
+
+    if (value < minValue || value > maxValue) {
```

Duplication can be further reduced.
Notice the common things you have in each function:
```js
if (/* the validity condition */) {
  displayError(fieldId);
} else {
  hiddenError(fieldId);
}
```
You can have a single function for displaying/hiding the error message that contains the if-else check. You can pass the boolean statement as one of the arguments.